### PR TITLE
chore: Re-structure Website

### PR DESCRIPTION
**Pull request by MichaReiser**
This PR restructures our Website by:

* Removing the dead careers link
* Moving the API documentation into the footer
* Splitting out a new `Docs` page (inspired by Tailwind, Parcel, Eslint, ...)

Improving the layout by

* Using the ~~`--primary-color`~~ `--primary-color` in dark mode and the `color3` in light mode for the active link in the TOC. `--primary-color` doesn't work in the light mode because it has too little contrast. 
* Using `font-weight: bold` for the main titles in the TOC

The PR includes some fixes:
* Enable the TOC for blog posts (requires change the header to `h2` and adding an id.
* Add the `DESCRIPTION_END` marker to only render the header of the rome formatter release blog post in the overview

I further made some improvements to the content.

## Long term

I think we want to move to a multi-page documentation where we have a page per section. This gives us more layouting option (and e.g. allows to directly link to the lint rules in the docs TOC). However, I don't know how to achieve this with our current website easily.

### CODE REVIEW

**Comment by ematipico** [comment]
I can see that the "Getting Started" section is repeated in the home page and the "Docs" page. Is it intentional?

File: website/src/_includes/docs/project-config.md [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
-
-We are deliberately lean with the supported configuration. 
-We do not include options just for the sake of personalization. 
-We aim to offer everything out of the box and only introduce configuration if [absolutely necessary](https://rome.tools/#philosophy).
+Here's an example:
 
 ```json
 {
   "formatter": {
+		"enabled": true,
```
Comment (ematipico) [inline website/src/_includes/docs/project-config.md:14]: Wrong indentation?

File: website/src/_includes/docs/reference.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
+## Reference
```
Comment (ematipico) [inline website/src/_includes/docs/reference.md:1]: What's "Reference" mean here? Why not CLI instead?

File: website/src/_includes/docs/reference.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+> Default: `true`
+
+
+##### `linter.rules.[category]`
+
+Options that influence the rules of a single category. Rome supports the following categories:
+
+* `correctness`: Code that is wrong or useless
+* `style`: Code that should be written in a more idiomatic way
+* `nursery`: new rules that are still under development.
```
Comment (ematipico) [inline website/src/_includes/docs/reference.md:97]: Just a suggestion for a next PR, we could make this piece to include an external file - like `{{ brief_categories }}` -  and make that autogenerated when running `cargo lintdoc`. So if we happen to create a new group, that's automatically generated.

**Review by ematipico — APPROVED** [review]

**Comment by MichaReiser** [comment]
> I can see that the "Getting Started" section is repeated in the home page and the "Docs" page. Is it intentional?

Yes it's intentional but ideally, it shouldn't. The way I envision it is to have a short intro at the top of the page that has a call to action button "Get started". But that requires new content for the start page that I don't have right now. Do you know of a "minimum effort" way I could accomplish the same without repeating?

File: website/src/_includes/docs/reference.md (hunk 3 above)
Comment (MichaReiser) [reply website/src/_includes/docs/reference.md:97]: That makes sense. I just don't think I have the time right now to invest into code gen. The same applies for our reference documentation that could be generated too.

File: website/src/_includes/docs/reference.md (hunk 2 above)
Comment (MichaReiser) [reply website/src/_includes/docs/reference.md:1]: It's the API reference of Rome and it includes the CLI and configuration schema. I got that idea from [Astrobuild](https://docs.astro.build/en/reference/configuration-reference/)

**Pull request merged**